bee-node/src/main.rs

```diff
@@ -36,41 +44,103 @@ async fn main() -> Result<(), Box<dyn Error>> {
     print_banner_and_version(true);
 
     // Deserialize the config.
-    let config = deserialize_config(cl_args);
+    let identity_path = cl_args.identity_path().unwrap_or(Path::new(IDENTITY_PATH)).to_owned();
+    let (identity_field, config) = deserialize_config(cl_args);
 
     // Initialize the logger.
     let logger_cfg = config.logger_config().clone();
     bee_common::logger::logger_init(logger_cfg)?;
 
+    // Establish identity.
+    let keypair = match read_keypair_from_pem_file(&identity_path) {
+        Ok(keypair) => {
+            if identity_field.is_some() {
+                warn!(
+                    "The config file contains an `identity` field which will be ignored. You may safely delete this field to suppress this warning."
+                );
+            }
+            keypair
+        }
+        Err(PemFileError::Read(_)) => {
+            // If we can't read from the file (which means it probably doesn't exist) we either migrate from the
+            // existing config or generate a new identity.
+            let keypair = if let Some(identity_encoded) = identity_field {
+                warn!(
+                    "There is no identity file at `{}`. Migrating identity from the existing config file.",
+                    identity_path.display(),
+                );
+
+                migrate_keypair(identity_encoded).unwrap_or_else(|e| {
+                    error!("Failed to migrate keypair: {}", e);
+                    std::process::exit(-1);
+                })
+            } else {
+                info!(
+                    "There is no identity file at `{}`. Generating a new one.",
+                    identity_path.display()
+                );
+
+                Keypair::generate()
+            };
+
+            if let Err(e) = write_keypair_to_pem_file(identity_path, &keypair) {
+                error!("Failed to write PEM file: {}", e);
+                std::process::exit(-1);
+            };
+
+            keypair
+        }
+        Err(e) => {
+            error!("Could not extract private key from PEM file: {}", e);
+            std::process::exit(-1);
+        }
+    };
+
+    let local = Local::from_keypair(keypair, config.alias().clone());
+
     // Start running the node.
     if config.run_as_entry_node() {
-        start_entrynode(config).await;
+        start_entrynode(local, config).await;
     } else {
-        start_fullnode(config).await;
+        start_fullnode(local, config).await;
     }
 
     Ok(())
 }
 
-fn deserialize_config(cl_args: ClArgs) -> NodeConfig<Storage> {
-    #[cfg(feature = "rocksdb")]
-    let config = match NodeConfigBuilder::<Storage>::from_file(
-        cl_args.config_path().unwrap_or_else(|| Path::new(CONFIG_PATH)),
-    ) {
+#[derive(Debug, thiserror::Error)]
+enum MigrationError {
```

Comment:
Please correct my older Self with this PR, and change the variants of this enum to `DecodeHex`, `DecodeKeypair`.